packages/nodes-base/nodes/Schedule/ScheduleTrigger.node.ts

```diff
@@ -430,7 +431,24 @@ export class ScheduleTrigger implements INodeType {
 				Second: moment.tz(timezone).format('ss'),
 				Timezone: moment.tz(timezone).format('z Z'),
 			};
-			this.emit([this.helpers.returnJsonArray([resultData])]);
+			const weeklyExecution = i ? (staticData[i] as IDataObject) : ({} as IDataObject);
+
+			// Checks if have the right week interval, handels new years as well
+			if (
+				isWeekly &&
+				weeklyExecution &&
+				i &&
+				(moment.tz(timezone).month() - (weeklyExecution.weekInterval as number) ===
+					(weeklyExecution.week as number) ||
+					moment.tz(timezone).month() + 52 - (weeklyExecution.weekInterval as number) ===
```

Comment:
Why 52? Please leave a comment explaining why this is needed

Reply:
52 is the total number of weeks, it's to cover the new year change if the last execution was on the 49th week with a 4 week interval if we are on the 1 week it will still trigger correctly.